js/app-page-settings.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * A page settings class for our application - an app running in the browser or in a webview
+ * only ever has one user, never multiple concurrent users, so we can defined its
+ * class here, and then export a singleton pageSettings for shared use in all pages.
+ *
+ * PageSettings is used to preserve state so when users go back to a project list view from
+ * a detail project card view they can be sent to where they were before
+ * (same loaded projects, same scroll position).
+ */
+class PageSettings {
+  constructor() {
+    this.reset();
+  }
+
+  reset() {
+    this.currentList = {};
+    this.currentScrollPosition = 0;
+    this.restore = false;
+  }
+
+  setCurrentList(list = {}) {
+    this.currentList = list;
+  }
+
+  setScrollPosition() {
+    if (typeof window !== `undefined`) {
+      this.currentScrollPosition = window.pageYOffset;
+    }
+  }
+
+  setRestore() {
```

Comment:
it's a future-proof in case we ever need to do more than just setting a property to true. But in that case this should really be
```
setRestore(value) {
  value = typeof value === "undefined"? false : value;
  this.shouldRestore = value;
}
```
since a `set` function should let the user specify which value should actually be used.